src/implementation/density_caching.rs

```diff
@@ -205,42 +206,44 @@ where
-        } else if z == -1 {
-            debug_assert!(x >= 0 && x <= subs as isize && y >= 0 && y <= subs as isize);
-            let index = 4 * face_size + (subs + 1) * x as usize + y as usize;
-            return self.regular_cache_extended[index];
-        } else if z == subs as isize + 1 {
-            debug_assert!(x >= 0 && x <= subs as isize && y >= 0 && y <= subs as isize);
-            let index = 5 * face_size + (subs + 1) * x as usize + y as usize;
-            return self.regular_cache_extended[index];
-        } else {
-            debug_assert!(
-                x >= 0
-                    && x <= subs as isize
-                    && y >= 0
-                    && y <= subs as isize
-                    && z >= 0
-                    && z <= subs as isize
-            );
-            let index = self.regular_block_index(x, y, z);
-            return self.regular_cache[index];
+
+        if 0 < self.regular_cache_extended.len() {
+            if x == -1 {
+                debug_assert!(y >= 0 && y <= subs as isize && z >= 0 && z <= subs as isize);
+                let index = 0 * face_size + (subs + 1) * y as usize + z as usize;
+                return self.regular_cache_extended[index];
+            } else if x == subs as isize + 1 {
+                debug_assert!(y >= 0 && y <= subs as isize && z >= 0 && z <= subs as isize);
+                let index = 1 * face_size + (subs + 1) * y as usize + z as usize;
+                return self.regular_cache_extended[index];
+            } else if y == -1 {
+                debug_assert!(x >= 0 && x <= subs as isize && z >= 0 && z <= subs as isize);
+                let index = 2 * face_size + (subs + 1) * x as usize + z as usize;
+                return self.regular_cache_extended[index];
+            } else if y == subs as isize + 1 {
+                debug_assert!(x >= 0 && x <= subs as isize && z >= 0 && z <= subs as isize);
+                let index = 3 * face_size + (subs + 1) * x as usize + z as usize;
+                return self.regular_cache_extended[index];
+            } else if z == -1 {
+                debug_assert!(x >= 0 && x <= subs as isize && y >= 0 && y <= subs as isize);
+                let index = 4 * face_size + (subs + 1) * x as usize + y as usize;
+                return self.regular_cache_extended[index];
+            } else if z == subs as isize + 1 {
+                debug_assert!(x >= 0 && x <= subs as isize && y >= 0 && y <= subs as isize);
+                let index = 5 * face_size + (subs + 1) * x as usize + y as usize;
+                return self.regular_cache_extended[index];
+            }
         }
+        debug_assert!(
+            x >= 0
+                && x <= subs as isize
+                && y >= 0
+                && y <= subs as isize
+                && z >= 0
+                && z <= subs as isize
+        );
+        let index = self.regular_block_index(x, y, z);
+        return self.regular_cache[index];
```

Comment:
At this point the failure is obvious: a transition side is used where it shouldn't. 
Unfortunately I don't quite understand the failed assertion, but it if more  informative, than the previous. 
I would suggest adding a message here, e.g. like so: 
```suggestion
        debug_assert!(
            x >= 0
                && x <= subs as isize
                && y >= 0
                && y <= subs as isize
                && z >= 0
                && z <= subs as isize,
            "An incorrect transition side was added to an extraction!"
        );
        let index = self.regular_block_index(x, y, z);
        return self.regular_cache[index];
```